refinery/core/views.py

```diff
@@ -754,20 +755,16 @@ def get(self, request, uuid):
 
     def patch(self, request, uuid):
         node = self.get_object(uuid)
-        old_file_uuid = node.file_uuid
+        new_file_uuid = request.data.get('file_uuid')
         data_set_owner = node.study.get_dataset().get_owner()
 
         if data_set_owner == request.user:
-            serializer = NodeSerializer(node, data=request.data, partial=True)
-            if serializer.is_valid():
-                self.update_file_store(old_file_uuid, node.file_uuid)
-                NodeIndex().update_object(node, using="data_set_manager")
-                serializer.save()
-                return Response(
-                    serializer.data, status=status.HTTP_202_ACCEPTED
-                )
+            # to remove the data file, we need to delete it and update index,
+            #  the file store item uuid should remain
+            self.update_file_store(node.file_uuid, new_file_uuid)
+            NodeIndex().update_object(node, using="data_set_manager")
             return Response(
-                serializer.errors, status=status.HTTP_400_BAD_REQUEST
+                NodeSerializer(node).data, status=status.HTTP_202_ACCEPTED
```

Comment:
All the operations above are synchronous, so HTTP 200 status is appropriate here.

Reply:
Also, is `file_uuid` actually updated in the database?

Reply:
@hackdna No, see pull request comments above.